src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -46,6 +46,15 @@ interface IncomingPaymentsDb {
 
     /** List received payments (with most recent payments first). */
     suspend fun listReceivedPayments(count: Int, skip: Int, filters: Set<PaymentTypeFilter> = setOf()): List<IncomingPayment>
+
+    /** List incoming payments (with the most recent payments first). */
+    suspend fun listIncomingPayments(count: Int, skip: Int, filters: Set<PaymentTypeFilter> = setOf()): List<IncomingPayment>
+
+    /** List expired unpaid normal payments created within specified time range (with the most recent payments first). */
+    suspend fun listExpiredPayments(fromCreatedAt: Long = 0, toCreatedAt: Long = currentTimestampMillis()): List<IncomingPayment>
+
+    /** Remove a pending incoming payment.*/
+    suspend fun removeIncomingPayment(paymentHash: ByteVector32)
```

Comment:
What about returning a boolean here to indicate whether the payment was removed or not?
This would let us return `false` for paid (or partially paid) payments.

Reply:
fixed in [be38964](https://github.com/ACINQ/lightning-kmp/pull/322/commits/be38964082f80475adeb7ff0f2bd9455988f5afb); I also only count successfully removed payments in the `IncomingPaymentsHandler`.

Reply:
On second thought, `listExpiredPayments` will not return paid (or partially paid) incoming payments so it should be a rare race condition to have on that list expired payments that were [partially] paid. It is still worth keeping this change though for the case where `removeIncomingPayment` is used for something where an absent payment is more likely (or important).

Reply:
Agreed, we will probably never see it return `false` because we're using it correctly, but it makes sense to keep that boolean for people who may use this function differently in the future.